packages/gen-ai-hub/src/utils/deployment-resolver.ts

```diff
@@ -1,19 +1,65 @@
-import {
-  DeploymentApi,
-  AiDeployment,
-  AiDeploymentStatus
-} from '@sap-ai-sdk/ai-core';
+import { DeploymentApi, AiDeployment } from '@sap-ai-sdk/ai-core';
+
+/**
+ * The deployment configuration when using a model. It can be either the name of the model or an object containing the name and version of the model.
+ * @typeParam ModelNameT - String literal type representing the name of the model.
+ */
+export type ModelConfiguration<ModelNameT = string> =
+  | ModelNameT
+  | {
+      /**
+       * The name of the model.
+       */
+      modelName: ModelNameT;
+      /**
+       * The version of the model.
+       */
+      modelVersion?: string;
+    };
+
+/**
+ * The deployment configuration when using a deployment ID.
+ */
+export interface DeploymentIdConfiguration {
+  /**
+   * The deployment ID.
+   */
+  deploymentId: string;
+}
+
+/**
+ * The deployment configuration can be either a model configuration or a deployment ID configuration.
+ * @typeParam ModelNameT - String literal type representing the name of the model.
+ */
+export type DeploymentConfiguration<ModelNameT = string> =
+  | ModelConfiguration<ModelNameT>
+  | DeploymentIdConfiguration;
+
+/**
+ * Type guard to check if the given deployment configuration is a deployment ID configuration.
+ * @param deploymentConfig - Configuration to check.
+ * @returns `true` if the configuration is a deployment ID configuration, `false` otherwise.
+ */
+export function isDeploymentIdConfiguration(
+  deploymentConfig: DeploymentConfiguration
+): deploymentConfig is DeploymentIdConfiguration {
+  return (
+    typeof deploymentConfig === 'object' && 'deploymentId' in deploymentConfig
```

Comment:
[q] How does this work, considering the payload actually contains `id`?

Reply:
I think you are referring to a different type `DeploymentConfiguration` is the parameter type and this function is a type guard that checks whether it is the "subtype" `DeploymentIdConfiguration`.

Reply:
ah okay never mind, I see that in a different place the `id` is extracted from the full response.